src/neo4j_genai/filters.py

```diff
@@ -0,0 +1,349 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Any, Type
+from collections import Counter
+
+
+DEFAULT_NODE_ALIAS = "node"
+
+
+class Operator:
+    """Operator classes are helper classes to build the Cypher queries
+    from a filter like {"field_name": "field_value"}
+    They implement two important methods:
+    - lhs: (left hand side): the node + property to be filtered on
+        + optional operations on it (see ILikeOperator for instance)
+    - cleaned_value: a method to make sure the provided parameter values are
+        consistent with the operator (e.g. LIKE operator only works with string values)
+    """
+
+    CYPHER_OPERATOR = None
+
+    def __init__(self, node_alias=DEFAULT_NODE_ALIAS):
+        self.node_alias = node_alias
+
+    def lhs(self, field):
+        return f"{self.node_alias}.`{field}`"
```

Comment:
1. Prefix any field name backtick with an backtick
2. If you find non alphabetic characters in the field name, wrap the whole field name with backticks

```
name -> name
1name -> `1name`
na`me -> `na``me`
```

Example of valid Cypher:

```
MATCH (n:`Hell``o`) WHERE n.`prop``erty` RETURN n LIMIT 25
```

More info: https://neo4j.com/docs/cypher-manual/current/syntax/naming/#symbolic-names-escaping-rules

Reply:
Interesting, thanks! Wdyt about always wrapping the field name in backticks, whether it contains special characters or not? In your example it means that 
``` 
name -> `name`
```

Reply:
I'd see that we'd rather not do that. It makes the queries more noisy when debugging.

Reply:
I understand your point, and at the same time I'm a bit scared that we miss some special characters and produce invalid query "just" to make it nice. Let me check if we can find something really robust.

Reply:
See here, if you think the test cases are covering enough use cases: https://github.com/neo4j/neo4j-genai-python/pull/29/commits/2b89aff3865e3987c89bf5de0947288559c21c70#diff-00b9f6996b286b97ed1b966f8596505071f2876cac256d97e31bb0de4449b198R53-R69

Reply:
Yeah, I see your concern. But if anyone should be able to do the right thing it should be us :) 
This should give us confidence: https://github.com/neo4j/cypher-builder/blob/main/src/utils/escape.ts#L54-L63

Reply:
Fair enough, used the same regex to test the field validity.